nomos/raw/p2p-network.md

```diff
@@ -0,0 +1,278 @@
+---
+title: P2P-NETWORK
+name: Nomos P2P Network Specification
+status: draft
+category: networking
+tags: [p2p, networking, libp2p, kademlia, gossipsub, quic]
+editor: Daniel Sanchez-Quiros <danielsq@status.im>
+contributors:
+- Filip Dimitrijevic <filip@status.im>
+---
+
+## Abstract
+
+This specification defines the peer-to-peer (P2P) network layer for Nomos blockchain nodes. The network serves as the comprehensive communication infrastructure enabling transaction dissemination through mempool and block propagation. The specification leverages established libp2p protocols to ensure robust, scalable performance with low bandwidth requirements and minimal latency while maintaining accessibility for diverse hardware configurations and network environments.
+
+## Motivation
+
+The Nomos blockchain requires a reliable, scalable P2P network that can:
+
+1. **Support diverse hardware**: From laptops to dedicated servers across various operating systems and geographic locations
+2. **Enable inclusive participation**: Allow non-technical users to operate nodes with minimal configuration
+3. **Maintain connectivity**: Ensure nodes remain reachable even with limited connectivity or behind NAT/routers
+4. **Scale efficiently**: Support large-scale networks (+10k nodes) with eventual consistency
+5. **Provide low-latency communication**: Enable efficient transaction and block propagation
+
+## Specification
+
+### Network Architecture Overview
+
+The Nomos P2P network addresses three critical challenges:
+
+- **Peer Connectivity**: Mechanisms for peers to join and connect to the network
+- **Peer Discovery**: Enabling peers to locate and identify network participants  
+- **Message Transmission**: Facilitating efficient message exchange across the network
+
+### Transport Protocol
+
+#### QUIC Transport
+
+The Nomos network employs **QUIC** as the primary transport protocol, leveraging the libp2p implementation.
+
+**Rationale for QUIC:**
+
+- Rapid connection establishment
+- Enhanced NAT traversal capabilities (UDP-based)
+- Built-in multiplexing simplifies configuration
+- Production-tested reliability
+
+**Protocol Identifier:**
+
+```text
+/quic-v1
+```
```

Comment:
We use quic but it is not really a protocol itself. We should remove the identifier here.